beetsplug/replaygain.py

```diff
@@ -1312,21 +1333,35 @@ def handle_album(self, album, write, force=False):
             discs[1] = album.items()
 
         for discnumber, items in discs.items():
-            try:
-                album_gain = self.backend_instance.compute_album_gain(
-                    items, target_level, peak
-                )
+            def _store_album(album_gain):
                 if len(album_gain.track_gains) != len(items):
                     raise ReplayGainError(
                         u"ReplayGain backend failed "
                         u"for some tracks in album {0}".format(album)
                     )
+                try:
+                    for item, track_gain in zip(items,
+                                                album_gain.track_gains):
+                        store_track_gain(item, track_gain)
+                        store_album_gain(item, album_gain.album_gain)
+                        if write:
+                            item.try_write()
+                        self._log.debug(u'done analyzing {0}', item)
+                except ReplayGainError as e:
+                    self._log.info(u"ReplayGain error: {0}", e)
+                except FatalReplayGainError as e:
+                    raise ui.UserError(u"Fatal ReplayGain error: {0}".format(e))
```

Comment:
I’m not entirely clear about why we need these exception handlers. Aren’t we just storing the data at this point, rather than doing any analysis? So is it really possible for this code to raise these exceptions?

Reply:
You're right, my bad.